backend/siarnaq/bracket/challonge.py

```diff
@@ -126,21 +126,118 @@ def get_tournament_data(tournament: Tournament, *, is_private: bool):
+
+    return data
 
-def get_round_indexes(tournament: Tournament, *, is_private: bool):
+
+# TODO should this be private-only, and thus should I drop the is_private flag?
+# Could use help/opinions
+def _get_round_indexes(tournament: Tournament, *, is_private: bool):
+    """
+    Returns round indexes of the tournament,
+    in order of Challonge's suggested play order.
+    """
     tournament_data = get_tournament_data(tournament, is_private=is_private)
 
-    round_indexes = set()
-    for item in tournament_data["included"]:
-        match item:
-            case {"type": "match", "attributes": {"round": round_index}}:
-                round_indexes.add(round_index)
+    round_indexes = list()
+
+    matches = [item for item in tournament_data["included"] if item["type"] == "match"]
+    matches.sort(key=lambda i: i["attributes"]["suggestedPlayOrder"])
+
+    for match in matches:
+        round_index = match["attributes"]["round"]
+        if round_index not in round_indexes:
+            round_indexes.append(round_index)
 
     return round_indexes
 
 
+def _get_round_indexes_and_names(tournament: Tournament, *, is_private=bool):
+    """
+    Based on empirical observation and design principles,
+    we declare any losers' rounds to be part of the same "overall" round
+    that
+    """
+    from siarnaq.api.episodes.models import TournamentStyle
+
+    round_indexes = _get_round_indexes(tournament, is_private=is_private)
+
+    round_indexes_and_names = []
+
+    if tournament.style == TournamentStyle.SINGLE_ELIMINATION:
+        for round_index in round_indexes:
+            round_indexes_and_names.append((round_index, f"Round {round_index}"))
+        return round_indexes_and_names
+
+    # Otherwise, we are in double elimination
+    current_winners_round_number = None
+    for position in range(len(round_indexes)):
+        round_index = round_indexes[position]
+
+        if round_index > 0:
+            current_winners_round_number = round_index
+            round_type = "(Winners)"
+        else:
+            # We are processing losers' rounds.
+            # Use the same round "number" as the most recent winners round.
+            is_next_round_winners = round_indexes[position + 1] > 0
+
+            if is_next_round_winners:
+                # The next round is a winners' round
+                # so this round is a "major" losers' round.
+                round_type = "(Losers Major)"
+            else:
+                # The next round is a losers' round,
+                # so this round is a "minor" losers' round.
+                round_type = "(Losers Minor)"
+
+        round_indexes_and_names.append(
+            (round_index, f"Round {current_winners_round_number} {round_type}")
+        )
+
+    return round_indexes_and_names
+
+
+# TODO should this be private-only, and thus should I drop the is_private flag?
+# Could use help/opinions
+def create_round_objects(tournament: Tournament, *, is_private: bool):
```

Comment:
similar question here too. note that this method is _only_ called w the private bracket (for now...)

Reply:
I think for this function, doesn't make sense to keep `is_private` as a kwarg, because creating round objects is an operation which intuitively operates on the tournament, not private/public brackets.